langlib/lang.transaction/src/main/ballerina/transaction.bal

```diff
@@ -91,6 +135,21 @@ public transactional isolated function getRollbackOnly() returns boolean = @java
 
 # Associates some data with the current transaction branch.
 #
+# ```
+# readonly & any|error data = {"author": "John", "fruits": ["Apple", "Orange"]};
+#
+# function createEntity() {
+#     transaction {
+#         transaction:setData(data);
```

Comment:
Let's make it inline with real-world data and add output. Update the next sample accordingly

```suggestion
#         transaction:setData({accessType: "RO"});
#         transaction:getData() ⇒ {"accessType":"RO"}
```